src/net/smooth/segment_loader.ts

```diff
@@ -119,17 +116,19 @@ const generateSegmentLoader = (
       responseData = new Uint8Array(0);
     }
 
-    return Observable.of({
+    return observableOf({
       type: "data" as "data", // :/ TS Bug or I'm going insane?
       value: { responseData },
     });
   }
+  else if (segment.mediaURL == null) {
+    return observableOf({
+      type: "data" as "data",
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=254&issues=AWRfEDcE1IJZDfsS9yTp&open=AWRfEDcE1IJZDfsS9yTp)

[ISSUE_KEY]: AWRfEDcE1IJZDfsS9yTp